src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -122,75 +140,212 @@ public static function get_core_data() {
 	/**
 	 * Returns the theme's data.
 	 *
-	 * Data from theme.json can be augmented via the $theme_support_data variable.
-	 * This is useful, for example, to backfill the gaps in theme.json that a theme
-	 * has declared via add_theme_supports.
-	 *
-	 * Note that if the same data is present in theme.json and in $theme_support_data,
-	 * the theme.json's is not overwritten.
+	 * Data from theme.json will be backfilled from existing
+	 * theme supports, if any. Note that if the same data
+	 * is present in theme.json and in theme supports,
+	 * the theme.json takes precendence.
 	 *
 	 * @since 5.8.0
+	 * @since 5.9.0 Theme supports have been inlined and the argument removed.
 	 *
-	 * @param array $theme_support_data Optional. Theme support data in theme.json format.
-	 *                                  Default empty array.
 	 * @return WP_Theme_JSON Entity that holds theme data.
 	 */
-	public static function get_theme_data( $theme_support_data = array() ) {
+	public static function get_theme_data() {
```

Comment:
Method params like this should not be removed for backwards compat reasons. Do something like this 

```suggestion
	public static function get_theme_data( $deprecated = array() ) {
	      if ( ! empty( $deprecated ) ) {
		 _deprecated_argument( __FUNCTION__, '5.9' );
	      }
```

Reply:
See my comment above https://github.com/WordPress/wordpress-develop/pull/1808#discussion_r744683229 as this class being marked as `@access private`, ie for internal core usage only.